src/svg.c

```diff
@@ -26,11 +26,12 @@ int svg_to_ps(lua_State *L) {
   }
   struct NSVGimage* image;
   image = nsvgParse((char*)input, "pt", em);
-  int max_output = 256;
+  int max_output = 512;
   int output_l = 0;
   char *output = malloc(max_output);
   output[0] = '\0';
   for (NSVGshape *shape = image->shapes; shape != NULL; shape = shape->next) {
+    char* strokeFillOper = "s"; // Just stroke
     for (NSVGpath *path = shape->paths; path != NULL; path = path->next) {
```

Comment:
I'm suspicious of `safe_append()` above.

1. `strlen()` is the length of the string *without* the null terminator, so it seems like line 12 should have a `>=` to accommodate it (because `strncat()` adds it). That could result in a missing character, though, rather than the added garbage in the test failure.
2. More importantly, the limit in `strncat()` is supposed to be the number of characters to append, not the total size of the buffer. So the function is effectively functioning as `strcat()` here. 

I think together these problems mean that `safe_append()` can write a null character into the byte stored just past whatever `output` points to, which could explain the selective test failure. But it's hard to analyze these problems accurately. 

I would store the result of `strlen(s2)` before the first call, use it on line 12 but with the `>=`, pass the value to `strncat()` and then increment output_l by the value instead of calling `strlen()` again. Alternatively you could just call `strcat()` given that the code is already ensuring the buffer is large enough.